ui/src/reusable_ui/types/index.ts

```diff
@@ -16,10 +16,10 @@ export enum ComponentSize {
 }
 
 export enum ComponentStatus {
```

Comment:
This name feels a little too general to me. What do you think about something like `AsyncControlStatus`? or `AsyncInputStatus`?

Reply:
I am planning on using it in as many components as possible, so they all can be displayed in a consistent set of states. It's not always related to async either, but can be. I have it named similarly to `ComponentColor` and `ComponentSize` as most of the reusable stuff uses those as well. I am open to a less general naming scheme, I just want to make sure it works for all 3 of these